.github/workflows/documentation-tests.yaml

```diff
@@ -6,19 +6,17 @@ on:
         description: 'Documentation branch name'
         required: true
         type: string
-        default: 'master'
+        default: '3.0.x-devel'
 
   push:
     branches:
-      - 'master'
+      - '3.0.x-devel'
     paths-ignore:
       - '**.md'
       - '**.txt'
       - '!**/CMakeLists.txt'
 
   pull_request:
-    branches:
-      - 'master'
```

Comment:
If a branch it's specified, the job is triggered only when a pull request is open to be merged in that specific branch. That is the reason why in your branch (Branch-out 3.0.x) it didn't run, so this must be changed in master (and it will be changed) so that every pull request will trigger this job. In our case it would be the same because this file will be inluded only in the pull requests that target 3.0.x-devel, but I think it makes more sense to not specify the branch to be more consistent with the future version of the master